tests/Doctrine/Tests/DBAL/Functional/MasterSlaveConnectionTest.php

```diff
@@ -189,4 +190,33 @@ public function testMasterSlaveConnectionCloseAndReconnect()
         $conn->connect('master');
         self::assertTrue($conn->isConnectedToMaster());
     }
+
+    public function testQueryOnMaster()
+    {
+        $conn = $this->createMasterSlaveConnection();
+
+        $query = 'SELECT count(*) as num FROM master_slave_table';
+
+        $statement = $conn->query($query);
+
+        self::assertTrue($statement instanceof Statement);
```

Comment:
This assertion seems redundant. The `query()` method always returns a statement.